lib/v2/shiep/index.js

```diff
@@ -37,30 +39,46 @@ module.exports = async (ctx) => {
 
     const host = `https://${type}.shiep.edu.cn`;
     const id = ctx.params.id || config[type].id;
+    const pubDateClass = type === 'dxxy' ? 'div[class="article-publishdate"]' : 'span[class="Article_PublishDate"]'; // 适配电子与信息工程学院网站改版
 
-    const response = await got({
-        method: 'get',
-        url: `${host}/${id}/list.htm`,
-    });
-    const data = response.data;
+    const response = await got(`${host}/${id}/list.htm`);
 
-    const $ = cheerio.load(data);
-    const className = 'list_item';
-    const pubDateClass = type === 'dxxy' ? 'div[class="article-publishdate"]' : 'span[class="Article_PublishDate"]'; // 适配电子与信息工程学院网站改版
+    const $ = cheerio.load(response.data);
 
-    const list = $(`.${className}`)
-        .get()
+    const items = $('.list_item')
+        .toArray()
         .filter((item) => {
             const date = dayjs($(item).find(pubDateClass).text().trim());
             return date.isValid();
+        })
+        .map((item) => {
+            item = $(item);
+            return {
+                title: item.find('a').attr('title') || item.find('a').text(),
+                link: new URL(item.find('a').attr('href'), host).href,
+                pubDate: parseDate(item.find(pubDateClass).text().trim(), 'YYYY-MM-DD'),
+            };
         });
 
-    const result = await util.ProcessFeed(host, list, pubDateClass, ctx.cache);
+    await Promise.all(
+        items.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                const response = await got(item.link);
+                const $ = cheerio.load(response.data);
+
+                item.description = art(path.join(__dirname, 'templates/description.art'), {
+                    description: $('.wp_articlecontent').html(),
+                });
```

Comment:
Some of the items will be redirected to the intranet, it would be better to leave a notice instead of `null`
![image](https://user-images.githubusercontent.com/11386903/213221071-70cd4caa-aeb6-490a-83bf-17dc50454548.png)